agent/agent.go

```diff
@@ -55,22 +55,27 @@ const (
-	shimPath             = flag.String("shim-path", "", "Path under which to handle websocket shim requests")
-	healthCheckPath      = flag.String("health-check-path", "/", "Path on backend host to issue health checks against.  Defaults to the root.")
-	healthCheckFreq      = flag.Int("health-check-interval-seconds", 0, "Wait time in seconds between health checks.  Set to zero to disable health checks.  Checks disabled by default.")
-	healthCheckUnhealthy = flag.Int("health-check-unhealthy-threshold", 2, "A so-far healthy backend will be marked unhealthy after this many consecutive failures. The minimum value is 1.")
-	disableGCEVM         = flag.Bool("disable-gce-vm-header", false, "Disable the agent from adding a GCE VM header.")
+	proxy                     = flag.String("proxy", "", "URL (including scheme) of the inverting proxy")
+	proxyTimeout              = flag.Duration("proxy-timeout", 60*time.Second, "Client timeout when sending requests to the inverting proxy")
+	host                      = flag.String("host", "localhost:8080", "Hostname (including port) of the backend server")
+	backendID                 = flag.String("backend", "", "Unique ID for this backend.")
+	debug                     = flag.Bool("debug", false, "Whether or not to print debug log messages")
+	disableSSLForTest         = flag.Bool("disable-ssl-for-test", false, "Disable requirements for SSL when running tests locally")
+	favIconURL                = flag.String("favicon-url", "", "URL of the favicon")
+	forwardUserID             = flag.Bool("forward-user-id", false, "Whether or not to include the ID (email address) of the end user in requests to the backend")
+	injectBanner              = flag.String("inject-banner", "", "HTML snippet to inject in served webpages")
+	bannerHeight              = flag.String("banner-height", "40px", "Height of the injected banner. This is ignored if no banner is set.")
+	shimWebsockets            = flag.Bool("shim-websockets", false, "Whether or not to replace websockets with a shim")
+	shimPath                  = flag.String("shim-path", "", "Path under which to handle websocket shim requests")
+	healthCheckPath           = flag.String("health-check-path", "/", "Path on backend host to issue health checks against.  Defaults to the root.")
+	healthCheckFreq           = flag.Int("health-check-interval-seconds", 0, "Wait time in seconds between health checks.  Set to zero to disable health checks.  Checks disabled by default.")
+	healthCheckUnhealthy      = flag.Int("health-check-unhealthy-threshold", 2, "A so-far healthy backend will be marked unhealthy after this many consecutive failures. The minimum value is 1.")
+	disableGCEVM              = flag.Bool("disable-gce-vm-header", false, "Disable the agent from adding a GCE VM header.")
+	enableWebsocketsInjection = flag.Bool("enable-websockets-injection", false, "Enables the injection of a header into a configured JSON path for all websocket messages.")
```

Comment:
s/all websocket/all text websocket/

This only makes sense for text messages; not binary ones.

Reply:
Correct me if I'm wrong, but can't binary websocket messages still decode properly to UTF-8 and be properly-formatted serialized JSON?